packages/shared/src/lib/share.ts

```diff
@@ -30,3 +30,47 @@ export const getTelegramShareLink = (link: string, text: string): string =>
   `https://t.me/share/url?url=${encodeURIComponent(link)}&text=${text}`;
 export const getEmailShareLink = (link: string, subject: string): string =>
   `mailto:?subject=${subject}&body=${encodeURIComponent(link)}`;
+
+export const getShareLink = (
+  provider: ShareProvider,
+  link: string,
+  text?: string,
+): string => {
+  switch (provider) {
+    case ShareProvider.WhatsApp:
+      return getWhatsappShareLink(link);
+    case ShareProvider.Twitter:
+      return getTwitterShareLink(link, text ?? '');
+    case ShareProvider.Facebook:
+      return getFacebookShareLink(link);
+    case ShareProvider.Reddit:
+      return getRedditShareLink(link, text ?? '');
+    case ShareProvider.LinkedIn:
+      return getLinkedInShareLink(link);
+    case ShareProvider.Telegram:
+      return getTelegramShareLink(link, text ?? '');
+    case ShareProvider.Email:
+      return getEmailShareLink(link, text ?? '');
+    default:
+      return link;
+  }
+};
+
+export const addLinkShareTrackingParams = (
+  link: string | undefined,
+  userId: string | undefined,
+  cid: string,
+): string => {
+  if (!link) {
+    return null;
+  }
+
+  if (!userId) {
+    return link;
+  }
+
+  const url = new URL(link);
+  url.searchParams.set('userId', userId);
+  url.searchParams.set('cid', cid);
```

Comment:
I know `cid` is not optional but it does not block someone to call it or being missed by some miss in types, so probably better to check `if (userid) searchParams.set`, or something similar.